packages/mco/components/disaster-recovery/applications/failover-and-relocate-modal/modal-body.tsx

```diff
@@ -0,0 +1,335 @@
+import * as React from 'react';
+import { useDeepCompareMemoize } from '@odf/shared/hooks/deep-compare-memoize';
+import { ApplicationModel } from '@odf/shared/models';
+import { ResourceIcon } from '@odf/shared/resource-link/resource-link';
+import { getName } from '@odf/shared/selectors';
+import { ApplicationKind } from '@odf/shared/types';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { referenceForModel } from '@odf/shared/utils';
+import {
+  useK8sWatchResources,
+  ObjectReference,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { Flex, FlexItem, Alert, AlertVariant } from '@patternfly/react-core';
+import { DR_SECHEDULER_NAME } from '../../../../constants';
+import {
+  ACMPlacementRuleModel,
+  ACMSubscriptionModel,
+  DRPlacementControlModel,
+} from '../../../../models';
+import {
+  ACMSubscriptionKind,
+  ACMPlacementRuleKind,
+  DRPlacementControlKind,
+} from '../../../../types';
+import {
+  matchApplicationToSubscription,
+  isPlacementRuleModel,
+} from '../../../../utils';
+import { DRPolicySelector } from './dr-policy-selector';
+import { PeerStatus } from './peer-status';
+import {
+  FailoverAndRelocateState,
+  FailoverAndRelocateAction,
+  FailoverAndRelocateType,
+  DRPolicyControlState,
+  ACTION_TYPE,
+  ErrorMessage,
+  ModalFooterStatus,
+} from './reducer';
+import { SubscriptionGroupSelector } from './subscription-group-selector';
+import { TargetClusterSelector } from './target-cluster-selector';
+import './modal-body.scss';
+
+export const findErrorMessage = (
+  errorMessage: ErrorMessage,
+  includeUpdateError: boolean = true
+) =>
+  [
+    errorMessage.drPolicyControlStateErrorMessage,
+    errorMessage.failoverAndRelocateActionErrorMessage,
+    errorMessage.targetClusterErrorMessage,
+    errorMessage.subscriptionGroupErrorMessage,
+    errorMessage.peerStatusErrorMessage,
+    includeUpdateError && errorMessage.failoverAndRelocateActionErrorMessage,
+  ]
+    .filter(Boolean)
+    .find((errorMessage) => errorMessage);
+
+const isObjectRefMatching = (
+  objectRef: ObjectReference,
+  objectRefList: string[]
+) => objectRefList?.includes(objectRef?.name);
+
+const filterDRPlacementRules = (
+  placementRules: ACMPlacementRuleKind[]
+): string[] =>
+  placementRules?.reduce(
+    (arr, placementRule) =>
+      placementRule?.spec?.schedulerName === DR_SECHEDULER_NAME
+        ? [...arr, getName(placementRule)]
+        : arr,
+    []
+  );
+
+const filterDRSubscriptions = (
+  application: ApplicationKind,
+  subscriptions: ACMSubscriptionKind[],
+  drPlacementRules: string[]
+): SubcsriptionMap =>
+  subscriptions?.reduce((obj, subscription) => {
```

Comment:
```suggestion
  subscriptions?.reduce((acc, subscription) => {
```
nit: stick to commonly used variable names with reduce such as `acc`, curr` or something relevant to make it easier to understand.